common/pkg/libartifact/store/store.go

```diff
@@ -174,19 +145,16 @@ func (as ArtifactStore) Pull(ctx context.Context, name string, opts libimage.Cop
 }
 
 // Push an artifact to an image registry.
-func (as ArtifactStore) Push(ctx context.Context, src, dest string, opts libimage.CopyOptions) (digest.Digest, error) {
-	if len(dest) == 0 {
-		return "", ErrEmptyArtifactName
-	}
-	destRef, err := alltransports.ParseImageName("docker://" + dest)
+func (as ArtifactStore) Push(ctx context.Context, src ArtifactStoreReference, dest ArtifactReference, opts libimage.CopyOptions) (digest.Digest, error) {
+	destRef, err := alltransports.ParseImageName("docker://" + dest.String())
```

Comment:
`destRef, err = docker.NewReference(dest.Named)` is a more efficient ~equivalent.

---

… both variants fail on a `repo:tag@digest` input, that is rejected by c/image. Generally, Podman, for Docker compatibility, allows such inputs, ignoring the tag. Do we want to do that here as well??? And what would use as an artifact name in the store?

My preference would be to reject such inputs, because most use cases are dubious (`repo:myCommentThatLooksLikeAVersionNumberButTheTagMayMoveIntheFuture@digestThatMatters`), but that might not be the right call.

_Either way_, `ArtifactReference` should document whether `repo:tag@digest` values are valid or not.

Reply:
> _Either way_, `ArtifactReference` should document whether `repo:tag@digest` values are valid or not.

… and if they are not valid, enforce that.